yb-voyager/src/query/queryissue/detectors.go

```diff
@@ -195,3 +203,90 @@ func (d *RangeTableFuncDetector) GetIssues() []QueryIssue {
 	}
 	return issues
 }
+
+type JsonConstructorFuncDetector struct {
+	query                                       string
+	unsupportedJsonConstructorFunctionsDetected mapset.Set[string]
+}
+
+func NewJsonConstructorFuncDetector(query string) *JsonConstructorFuncDetector {
+	return &JsonConstructorFuncDetector{
+		query: query,
+		unsupportedJsonConstructorFunctionsDetected: mapset.NewThreadUnsafeSet[string](),
+	}
+}
+
+func (j *JsonConstructorFuncDetector) Detect(msg protoreflect.Message) error {
+	switch queryparser.GetMsgFullName(msg) {
+	case queryparser.PG_QUERY_JSON_ARRAY_AGG_NODE:
+		j.unsupportedJsonConstructorFunctionsDetected.Add(JSON_ARRAYAGG)
+	case queryparser.PG_QUERY_JSON_ARRAY_CONSTRUCTOR_AGG_NODE:
+		j.unsupportedJsonConstructorFunctionsDetected.Add(JSON_ARRAY)
+	case queryparser.PG_QUERY_JSON_OBJECT_AGG_NODE:
+		j.unsupportedJsonConstructorFunctionsDetected.Add(JSON_OBJECTAGG)
+	case queryparser.PG_QUERY_JSON_OBJECT_CONSTRUCTOR_NODE:
+		j.unsupportedJsonConstructorFunctionsDetected.Add(JSON_OBJECT)
+	}
+	return nil
+}
+
+func (d *JsonConstructorFuncDetector) GetIssues() []QueryIssue {
+	var issues []QueryIssue
+	if d.unsupportedJsonConstructorFunctionsDetected.Cardinality() > 0 {
+		issues = append(issues, NewJsonConstructorFunctionIssue(DML_QUERY_OBJECT_TYPE, "", d.query, d.unsupportedJsonConstructorFunctionsDetected.ToSlice()))
+	}
+	return issues
+}
+
+type JsonQueryFunctionDetector struct {
+	query                                 string
+	unsupportedJsonQueryFunctionsDetected mapset.Set[string]
+}
+
+func NewJsonQueryFunctionDetector(query string) *JsonQueryFunctionDetector {
+	return &JsonQueryFunctionDetector{
+		query:                                 query,
+		unsupportedJsonQueryFunctionsDetected: mapset.NewThreadUnsafeSet[string](),
+	}
+}
+
+func (j *JsonQueryFunctionDetector) Detect(msg protoreflect.Message) error {
+	if queryparser.GetMsgFullName(msg) == queryparser.PG_QUERY_JSON_TABLE_NODE {
```

Comment:
can you add an example of JSON_TABLE parse tree. these if conditions are not clear.